smtp/smtpc.c

```diff
@@ -378,6 +378,7 @@ smtpc_task_execute(va_list list)
 int
 smtpc_execute(struct smtpc_request *req, double timeout)
 {
+	char curl_error[CURL_ERROR_SIZE];
```

Comment:
So we'll allocate it on the stack always, even when there is no any need. It is 256 bytes, not much, so should be okay if we'll not do it always: it is just waste of stack space. Allocation of arrays on a fiber stack (which is 512 KiB, not default 8 MiB) is generally discouraged: however 256 bytes should be okay.

Reply:
NB: We cannot convert it to the static array, because multiple fibers may enter into this function and yield on coio_call().

Reply:
I see now: we need to pass a buffer pointer to libcurl before start a request. So the reason above is not applicable. Sorry, I miss a concentration a bit tonight.

Anyway, we allocate it on the stack just to copy it into dynamic memory later. I think it would be better to allocate this within the request structure. This way we'll not need to do those copies. We'll not need to do the extra malloc() / strndup() below.

We also can pass `curl_error + sizeof("SMTP error: ")` to libcurl and fill the buffer start with those bytes when an error actually occurs. So we'll not need to do those concatenation using `snprintf()`.